crates/ruff_python_formatter/src/expression/expr_string_literal.rs

```diff
@@ -1,34 +1,41 @@
 use ruff_formatter::FormatRuleWithOptions;
-use ruff_python_ast::AnyNodeRef;
-use ruff_python_ast::ExprStringLiteral;
+use ruff_python_ast::{AnyNodeRef, ExprStringLiteral};
 use ruff_text_size::{Ranged, TextLen, TextRange};
 
 use crate::comments::SourceComment;
-use crate::expression::parentheses::{NeedsParentheses, OptionalParentheses};
-use crate::expression::string::{
-    AnyString, FormatString, StringLayout, StringPrefix, StringQuotes,
+use crate::expression::parentheses::{
+    in_parentheses_only_group, NeedsParentheses, OptionalParentheses,
 };
 use crate::prelude::*;
+use crate::string::{
+    AnyString, FormatStringContinuation, StringOptions, StringPrefix, StringQuotes,
+};
 
 #[derive(Default)]
 pub struct FormatExprStringLiteral {
-    layout: StringLayout,
+    options: StringOptions,
```

Comment:
It seems the comment here went lost for some reason. 

We could avoid using the generic `StringOptions` here by introducing an `ExprStringLiteralKind` that is either `String` or `Docstring` and do the same for `StringLiteral` but with an extra `InFString` variant. It may help future readers to better understand which invariants are possible (and from where the formattign is called). 

```rust
#[derive(Default)]
pub struct FormatExprStringLiteral {
    options: ExprStringLiteralKind,
}

#[derive(Default, Copy, Clone, Debug)]
pub enum ExprStringLiteralKind {
    #[default]
    String,
    Docstring,
}

impl ExprStringLiteralKind {
    const fn string_literal_kind(self) -> StringLiteralKind {
        match self {
            ExprStringLiteralKind::String => StringLiteralKind::String,
            ExprStringLiteralKind::Docstring => StringLiteralKind::Docstring,
        }
    }

    const fn is_docstring(self) -> bool {
        matches!(self, Self::Docstring)
    }
}


#[derive(Default)]
pub struct FormatStringLiteral {
    layout: StringLiteralKind,
}

#[derive(Copy, Clone, Debug, Default)]
pub enum StringLiteralKind {
    #[default]
    String,
    Docstring,
    InFString(Quoting),
}
```